Fleetish theme renamed to fleet dark and adjusted to match official theme.

krfl opened the pull request:
Fleetish was originally based on an early screenshot of JetBrains Fleet. Fleet's theme changed considerably between the initial reveal and the public beta/preview and while Fleetish does implement many of these changes, it also keeps elements of the old theme used in the screenshot.

While I never intended Fleetish to be a perfect copy of Fleet's official theme, I now see that the name Fleet-ish can be a little problematic because it sets the expectation of a theme similar to that of Fleet. Thus I have changed the name of the theme to "fleet_dark" and done some work to further match it with the official dark Fleet theme. There is still nitpicking to be done, but this is a big step towards the "expected" look. I don't think it's possible to match it perfectly due to the way JetBrains seem to change their highlighting scopes from filetype to filetype, but it just needs to be _good enough_. It's currently "optimized" for rust and markdown, so there will be bigger differences for other languages. 

Lastly, this will likely break a few configurations, but I do believe these changes are for the better to avoid further confusion and disappointments. 


**Opinions and suggestions are very much welcome.** 


Below you can see some screenshots comparing Helix and Fleet. I also included some python stuff where you can clearly see differences. 

![fleet_dark1](https://user-images.githubusercontent.com/2630397/205465862-96a46eda-d20a-4e1a-834b-3b70f7fd73bd.jpg)

![fleet_dark2](https://user-images.githubusercontent.com/2630397/205465868-de10f88d-abd8-4b46-87ab-624d2894a0a3.jpg)

![fleet_dark3](https://user-images.githubusercontent.com/2630397/205465873-a3b9f4b6-5079-4f25-9863-2d5419da1ff8.jpg)

![fleet_dark4](https://user-images.githubusercontent.com/2630397/205465879-f1ae01ef-84cd-4ec7-bfa6-406f4ab2e7b2.jpg)


Tagging @matoous, @archseer and @pascalkuthe for comments and thoughts.

archseer commented:
The renaming is fine since it's a one line fix on the user's end, we'll denote this in the breaking changes section of the release notes 👍🏻

archseer commented:
Second image: to style `Fixed`, `Auto` inside the enum you can use `@type.enum.variant`

krfl commented:
> Second image: to style `Fixed`, `Auto` inside the enum you can use `@type.enum.variant`

Thank you! I pushed a fix for this.

matoous commented on runtime/themes/fleet_dark.toml:
```diff
@@ -76,25 +82,25 @@
 "ui.text" = "light" # .focus / .info
```
**suggestion**: this could provide better highlighting for the pickers matching the picker of fleet.
```suggestion
"ui.text" = "light" # .info
"ui.text.focus" = { fg = "lightest", bg = "blue_accent" }
```

krfl replied:
Could you provide a screenshot of what this looks like? I'm currently away from my laptop

matoous replied:
Sure think, sorry. 

Here's how the picker looks currently:

![Screenshot 2022-12-04 at 22 33 18](https://user-images.githubusercontent.com/15747583/205517010-a744af98-d260-445d-b975-f083d9186896.png)

Here's how it would look like with the change above (which is actually incorrect):

![Screenshot 2022-12-04 at 22 33 56](https://user-images.githubusercontent.com/15747583/205517021-7da00819-818c-4dcb-a4f0-8a8914aab0fd.png)

The correct highlight color should be `#204474` which would look like this:

![Screenshot 2022-12-04 at 22 38 06](https://user-images.githubusercontent.com/15747583/205517049-377412d6-11ff-4dfa-a15e-394ca8a60de5.png)

And that matches fleet:

<img width="1552" alt="Screenshot 2022-12-04 at 22 38 16" src="https://user-images.githubusercontent.com/15747583/205517058-918dd7ac-5969-4e32-aded-537830e5cf0f.png">

I guess the same should be used for `ui.menu.selected`, currently looking like this:

<img width="420" alt="Screenshot 2022-12-04 at 22 44 32" src="https://user-images.githubusercontent.com/15747583/205517253-54994de2-e27f-4531-8b71-f7d6cecb23f6.png">

versus (in Fleet):

<img width="420" alt="Screenshot 2022-12-04 at 22 44 32" src="https://user-images.githubusercontent.com/15747583/205517267-e95aa3d0-bd4a-420f-bed0-7c784e4639ff.png">

I.e., maybe this would work:

```toml
...
"ui.menu.selected" = { fg = "lightest", bg = "dark_blue_accent" } # .selected
"ui.text.focus" = { fg = "lightest", bg = "dark_blue_accent" }

...

```

krfl replied:
Looks great! Good job. Can we reuse the "selected" color or is that too dark?

matoous replied:
If we are to follow Fleet strictly it uses 2 different color, the one configured right now for `selected` for selections and the one above for the menues.

krfl replied:
Alright. Maybe name it "focus" instead of accent, as accent colors are much brighter than anything else?

matoous replied:
Yes, sounds good 👍

archseer commented:
```
"keyword.control.import" = "cyan"
```

to get `from`, `import` as cyan keywords in Python/Rust

archseer commented:
Actually most keywords seem cyan (`if`/`else` etc)? Maybe `keyword.control` as `cyan` is enough, and you need a special setting only for `keyword.control.exception`

krfl commented:
> Actually most keywords seem cyan (`if`/`else` etc)? Maybe `keyword.control` as `cyan` is enough, and you need a special setting only for `keyword.control.exception`

Fixed. Also adjusted the color of builtins. It's pretty close for Rust now.

archseer reviewed (approved):
Looks good to me! Thanks for the changes :)
